bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/discovery/ShellyDiscoveryParticipant.java

```diff
@@ -162,13 +187,14 @@ public DiscoveryResult createResult(@Nullable ServiceInfo service) {
                 addProperty(properties, PROPERTY_DEV_TYPE, thingType);
                 addProperty(properties, PROPERTY_DEV_MODE, mode);
 
-                logger.debug("Adding Shelly thing, UID={}", thingUID.getAsString());
+                logger.debug("{}: Adding Shelly thing, UID={}", name, thingUID.getAsString());
                 return DiscoveryResultBuilder.create(thingUID).withProperties(properties)
                         .withLabel(name + " - " + address).withRepresentationProperty(name).build();
             }
-        } catch (NullPointerException e) {
-            logger.warn("Device discovery failed for device {}, IP {}, service={}: {}", name, address, name,
-                    e.getMessage(), e);
+        } catch (IOException | IllegalArgumentException | NullPointerException e) { // maybe some format description was
+                                                                                    // buggy
+            logger.info("{}", messages.get("discovery.failed", name, address, ShellyUtils.getMessage(e)));
+            logger.debug("{}: Exception {}", name, e.toString());
```

Comment:
```suggestion
            logger.debug("{}: Exception {}", name, e.getMessage());
```

Reply:
if I use getMessage8) I don't get the stack trace, that's what I'm looking for here

Reply:
why is
```
logger.debug("{}: Unable to initialize, retrying later: {}", thingName, e);
``
results in 
```
[ERROR] Code Analysis Tool has found: 
 4 error(s)! 
 9 warning(s) 
 10 info(s)
[ERROR] org.openhab.binding.shelly.internal.handler.ShellyBaseHandler.java:[158]
Count of placeholder(2) is not equal to count of parameter(1)
[ERROR] org.openhab.binding.shelly.internal.handler.ShellyBaseHandler.java:[325]
Count of placeholder(3) is not equal to count of parameter(2)
[ERROR] org.openhab.binding.shelly.internal.handler.ShellyBaseHandler.java:[392]
Count of placeholder(3) is not equal to count of parameter(2)
[ERROR] org.openhab.binding.shelly.internal.handler.ShellyBaseHand
```

whereas it gets accepted when I use e.toString(). AFAIK log4j has support for argument type Throwable

Reply:
changed